### ISSUE TITLE
A problem with the training of  the fusion module

### ISSUE THREAD

**Issue by N1cl0ve** [iwa-shi/DC_VIC#2]
Hi Mr. Iwai,

I read your DC_VIC work and code, thank you for your excellent work. 

I have a little problem with training of  the fusion module in src.models.subnet.vq_fusion_module.py. As described in the paper, the parameters of pretrain VQGAN decoder are fixed. In src.models.comp_model.hyperprior_dc_vic_model.py line 235, the VQGAN encoder does not participate in gradient calculation but the fusion module in line 282 does not have similar operations. Are there other operations while the set optimizer or something for fusion module to keep the parameters of the pretrain VQGAN decoder? Can you answer the question and provide some details in training the fusion module? Thank you!

Sincerely,
Niclove

**Comment by iwa-shi** [comment]
Hi @N1cl0ve ,

Thank you for reaching out and for your interest in our work.

In our training code, we ensure that the parameters of the pre-trained VQGAN decoder remain fixed by setting `requires_grad=False` for the VQGAN decoder. This allows the fusion module’s parameters to be updated while keeping the VQGAN decoder unchanged.

We are planning to release the detailed training code and instructions soon, likely within a week.

Thank you for your patience!

Best regards,
Shoma Iwai

**Comment by iwa-shi** [comment]
Hi @N1cl0ve ,

Sorry for the wait, and thank you for your patience.

We have now released the training code, configuration files, and documentation. Please feel free to open an issue if you encounter any problems or have any questions.

**Comment by N1cl0ve** [comment]
Thank you very much!
